runelite-client/src/main/java/net/runelite/client/plugins/opponentinfo/OpponentInfoOverlay.java

```diff
@@ -89,44 +109,174 @@ private OpponentInfoOverlay(
+			{
+				Widget raidOpponentWrapper = client.getWidget(303, 6);
+				Widget raidOpponentNameWidget = client.getWidget(303, 11);
+				Widget raidOpponentHealthWidget = client.getWidget(303, 20);
+
+				if (raidOpponentNameWidget != null)
+				{
+					useVanilla = false;
+
+					if (lastHealth != 0)
+					{
+						for (String boss : opponentInfoConfig.vanillaHPBarOverride().split(","))
+						{
+							if (WildcardMatcher.matches(boss.toLowerCase().trim(), raidOpponentNameWidget.getText().toLowerCase().trim()))
+							{
+								useVanilla = true;
+								break;
+							}
+						}
+					}
+				}
+
+				if (raidOpponentWrapper != null && (!useVanilla && opponentInfoConfig.mergeRaidOverlay()))
+				{
+					raidOpponentWrapper.setHidden(true);
+					shouldUnHide = false;
+				}
+
+				if (raidOpponentNameWidget != null && raidOpponentName == null)
+				{
+					raidOpponentName = raidOpponentNameWidget.getText();
+				}
+
+				if (raidOpponentHealthWidget != null && raidOpponentMaxHealth == 0)
+				{
+					raidOpponentMaxHealth = Integer.parseInt(raidOpponentHealthWidget.getText().split(" / ")[1].split(" \\(")[0]);
+				}
+
+				else if (raidOpponentHealthWidget != null && opponentInfoConfig.showVanillaPercentages())
+				{
+					raidOpponentHealthWidget.setText(lastHealth + " / " + raidOpponentMaxHealth + " (" + String.format("%.1f", (lastHealth * 100.0) / raidOpponentMaxHealth) + "%)");
+				}
+
+				raidOpponentHealth = lastHealth;
+
+				if (lastHealth == 0)
+				{
+					//if lastHealth is zero, either the Raid Health Bar is supposed to disappear,
+					//or the opponent is dead, so remove the raid opponent name and max health vars
+					raidOpponentName = null;
+					raidOpponentMaxHealth = 0;
+
+					if (raidOpponentWrapper != null)
+					{
+						//unhide so that initial values can be read again, but wait 7 seconds so that the hp bar isn't shown
+						//while the boss is dying. It's possible to cancel the unhide by changing shouldUnHide back to false
+						shouldUnHide = true;
+
+						scheduler.schedule(() ->
+						{
+							if (shouldUnHide)
+							{
+								raidOpponentWrapper.setHidden(false);
+							}
+						}, 7, TimeUnit.SECONDS);
+					}
+				}
+			});
+		}
+	}
+
+	public void unHideHPBar()
+	{
+		clientThread.invokeLater(() ->
+		{
+			Widget raidOpponentWrapper = client.getWidget(303, 6);
+			if (raidOpponentWrapper != null)
+			{
+				raidOpponentWrapper.setHidden(false);
+			}
+
+			raidOpponentName = null;
+			raidOpponentMaxHealth = 0;
+			raidOpponentHealth = 0;
+			shouldUnHide = true;
+		});
+	}
+
+
 	@Override
 	public Dimension render(Graphics2D graphics)
 	{
-		final Actor opponent = opponentInfoPlugin.getLastOpponent();
-
-		if (opponent == null)
+		if (useVanilla && opponentInfoConfig.mergeRaidOverlay())
 		{
-			opponentName = null;
 			return null;
 		}
 
-		if (opponent.getName() != null && opponent.getHealthScale() > 0)
+		final Actor opponent = opponentInfoPlugin.getLastOpponent();
+
+		if (opponent != null)
```

Comment:
Why is this guard clause reversed?

Reply:
When the vanilla bar and the opponentinfo are merged, the opponentinfo shouldn't disappear after 5 seconds, or when the opponent is null, since the variables are still known from the vanilla bar. When opponent is null, it shouldn't return yet, but rather go on and check if the raid variables are known. There is an additional gaurd clause which checks if the raid variables are null after this guard clause:
`		if ((opponent == null && (vanillaHealthBarMaxHealth == 0 || !opponentInfoConfig.mergeVanillaHealthBar())) ||
			opponentName == null)
		{
			opponentName = null;
			return null;
		}`